snowpack/src/build/process.ts

```diff
@@ -166,10 +166,7 @@ export async function addBuildFiles(state: BuildState, files: string[]) {
       }
     }
     const fileUrls = getUrlsForFile(f, config)!;
-    // Only push the first URL. In multi-file builds, this is always the JS that the
-    // CSS is imported from (if it exists). That CSS may not exist, and we don't know
-    // until the JS has been built/loaded.
-    allFileUrls.push(fileUrls[0]);
```

Comment:
This is the strangest change—this is breaking behavior, done very intentionally, but I am not sure why we’re just throwing away build output. Perhaps we need to find why this change was made.

Reply:
OK, in investigating this a little more, I found the root issue. We have an inconsistency in how we say load plugins should work, vs how they actually work.

Take this example: `.svelte` files allow outputs `['.js', '.css']`. They always output the first (JS), but may not always output the second (CSS). Internally, we are basically making a rule that because `.css` _may_ not exist at times, we should just never request it. Which is causing problems as of `3.1`.

The expected change is that we should try and build all possible files a plugin may output. But currently, the build step operates as if all files **WILL** generate a response, and the reality is that many responses should be missing. This will require a pretty big TypeScript change 😅. But I believe a sweeping type change is going to be less problematic than trying to make a minimal change here and misrepresenting what’s going on behind-the-scenes as we have been doing.